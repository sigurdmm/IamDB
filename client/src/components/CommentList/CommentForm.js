import {
  Formik,
  Form,
  Field,
  ErrorMessage,
} from 'formik';
import React from 'react';
import PropTypes from 'prop-types';

import './CommentForm.less';

const CommentForm = ({ onSubmit }) => <Formik
  initialValues={{ comment: '' }}
  onSubmit={({ comment }) => onSubmit(comment)}
  validate={(values) => {
    const errors = {};

    if (!values.comment || values.comment.length < 10) {
      errors.comment = `Comment cannot be less than 10 characters. Your's is ${values.comment.length}`;
    }

    return errors;
  }}>
  <Form className="commentform">
    <label htmlFor="comment" className="commentform__label">Comment</label>
    <Field
      className="commentform__comment"
      id="comment"
      name="comment"
      rows="10"
      component="textarea"/>
    <ErrorMessage className="error" name="comment" component="div"/>

    <button type="submit">Submit comment</button>
  </Form>
</Formik>;

CommentForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};

export default CommentForm;
