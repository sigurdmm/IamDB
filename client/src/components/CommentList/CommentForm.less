
.commentform {
  box-sizing: border-box;

  padding: 0.5em 0;

  &__label {
    display: block;
  }

  &__comment {
    box-sizing: border-box;
    display: block;

    margin: 1em 0;

    width: 100%;
  }

  button[type="submit"] {
    display: block;

    padding: 0.5em 1em;
  }
}
