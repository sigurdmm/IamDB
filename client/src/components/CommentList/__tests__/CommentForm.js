import { render, shallow } from 'enzyme';

import CommentForm from '../CommentForm';

describe('<CommentForm/>', () => {
  it('should render properly', () => {
    const wrapper = render(<CommentForm onSubmit={() => null}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should pass comment to parent on submit', () => {
    const spy = jest.fn();
    const wrapper = shallow(<CommentForm onSubmit={spy}/>);

    wrapper.find('.commentform').find('button').simulate('click');
    wrapper.update();
  });
});
