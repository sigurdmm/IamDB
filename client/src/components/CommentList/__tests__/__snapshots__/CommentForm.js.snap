// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentForm/> should render properly 1`] = `
<form
  class="commentform"
>
  <label
    class="commentform__label"
    for="comment"
  >
    Comment
  </label>
  <textarea
    class="commentform__comment"
    id="comment"
    name="comment"
    rows="10"
  />
  <button
    type="submit"
  >
    Submit comment
  </button>
</form>
`;
